Feature/sign in

Kryzhanivsky opened the pull request.

Kryzhanivsky committed:
sign-in page created

Kryzhanivsky committed:
Created sign-in component

Orestli commented on src/layouts/sign-in/index.tsx:
```diff
@@ -0,0 +1,73 @@
+import Button from '@components/button';
+import Input from '@components/input';
+import routes from '@config/routes';
+import { FormValues, initialValues, schema } from '@layouts/sign-in/utils';
```
Use relative path

Orestli commented on src/layouts/sign-in/index.tsx:
```diff
@@ -0,0 +1,73 @@
+          </div>
+          <div className="flex justify-between">
+            <label className="flex justify-center items-center">
+              <input
```
Take checkbox out in the separate component

Orestli reviewed (changes requested):
Add cursor pointer to checkbox

Orestli commented on src/layouts/sign-in/index.tsx:
```diff
@@ -0,0 +1,73 @@
+          Sign In
+        </h1>
+        <div className="flex flex-col gap-3 xs:gap-4 md:gap-6">
+          <div className="flex gap-5 flex-col ">
```
Unnecessary space

Kryzhanivsky committed:
Created a custom check box as a separate component

Kryzhanivsky committed:
All edits have been resolved:

	1. Relative path used;
	2. The checkbox was moved to a separate component;
	3. Unnecessary space deleted;

Orestli commented on src/components/checkbox/index.tsx:
```diff
@@ -0,0 +1,36 @@
+  'id'
+>;
+
+const Checkbox: React.FC<InputProps> = ({ ...props }) => (
```
Get the `className` from the spread and pass it to the parent class (via `twMerge`) so that styles can be overridden from the outside

Orestli commented on src/components/checkbox/index.tsx:
```diff
@@ -0,0 +1,36 @@
+  >
+    <input type="checkbox" {...props} className="peer hidden" />
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
```
Save the svg in `/public/icons/` and render it as a component. Example [here](https://github.com/Rostand-Corp/flitch-frontend/blob/main/src/layouts/sign-up/index.tsx#L9)

Orestli commented on src/layouts/sign-in/index.tsx:
```diff
@@ -51,15 +53,14 @@ const SignIn = () => {
                 className="h-6 w-6"
                 {...formik.getFieldProps('keepSignedIn')}
               />
               <p className="mx-5">Keep me Signed In</p>
```
I think `margin` should only be on the left: `ml-5`

Kryzhanivsky committed:
refactor: the 'check' svg saved in /public/icons/ and rendered as a component.

Kryzhanivsky committed:
refactor: margin of <p> element;

Orestli commented on src/components/checkbox/index.tsx:
```diff
@@ -10,26 +12,16 @@ type InputProps = RequiredKeys<
-        d="M4.5 12.75l6 6 9-13.5"
-      />
-    </svg>
+    <Check className="w-7 w-7 invisible peer-checked:visible" />
```
Duplicate `w`

Orestli commented on src/components/checkbox/index.tsx:
```diff
@@ -1,5 +1,7 @@
 import { memo } from 'react';
+import { twMerge } from 'tailwind-merge';
 
+import Check from '@public/images/check.svg';
```
Create a new folder called `/icons` and move the SVG

Kryzhanivsky committed:
fix: dublicated code

Kryzhanivsky committed:
refactor: A new folder /icons created. check.svg moved into /icons;

Orestli committed:
refactor: update <Checkbox /> styles

Orestli reviewed (approved):
LGTM
